js/text-validation.js

```diff
@@ -0,0 +1,52 @@
+const uploadForm = document.querySelector('.img-upload__form');
+const tagsInput = uploadForm.querySelector('.text__hashtags');
+const MAX_TAGS_NUMBER = 5;
+const MAX_TAGS_LENGTH = 20;
+const pattern = /[^a-zA-Z0-9]/;
+const description = uploadForm.querySelector('.text__description');
+const MAX_DESCRIPTION_LENGTH = 140;
+
+tagsInput.addEventListener('input', onTagsInputInput);
+
+function onTagsInputInput () {
+  const tags = tagsInput.value.split(' ');
+  for (let i=0; i < tags.length; i++) {
+    if (tags[i][0] !== '#') {
+      tagsInput.setCustomValidity('Теги должны начинаться с #')
+    } else if (tags.length > MAX_TAGS_NUMBER) {
+      tagsInput.setCustomValidity('Максимальное число тегов: ' + MAX_TAGS_NUMBER);
+    } else if (pattern.test(tags[i].slice(1))) {
+      tagsInput.setCustomValidity('Теги должны состоять только из букв и числел');
+    } else if (tags[i].length > MAX_TAGS_LENGTH) {
+      tagsInput.setCustomValidity('Слишком длинный тег. Максимум символов: ' + MAX_TAGS_LENGTH);
+    } else if (tags[i].length === 1) {
+      tagsInput.setCustomValidity('Тег не может быть пустым');
+    } else if (hasDuplicates(tags)) {
+      tagsInput.setCustomValidity('Удалите повторяющиеся теги');
+    } else {
+      tagsInput.setCustomValidity('');
+    }
+  }
+}
+
+function hasDuplicates (array) {
+  let valuesSoFar = Object.create(null);
```

Comment:
Прикольно, конечно, но 
```suggestion
  let valuesSoFar = {};
```
так проще.
Мне тут не принципиально, можешь оставить :)